### PR TITLE
Flagset cache flagsetfilter

### PR DESCRIPTION
**Pull request by chillaq**
# Ruby SDK

### CODE REVIEW

File: lib/splitclient-rb/cache/filter/flag_set_filter.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      class FlagSetsFilter
+        def initialize(flag_sets = [])
+          @flag_sets = Set.new(flag_sets)
+          @should_filter = @flag_sets.any?
+        end
+
+        def flag_set_exist?(flag_set)
+          if not @should_filter
+            return true
+          end
```
Comment (sanzmauro) [inline lib/splitclient-rb/cache/filter/flag_set_filter.rb:17]: you can replace this `if` for this line: ` return true unless @should_filter`

File: lib/splitclient-rb/cache/filter/flag_set_filter.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+          if not flag_set.respond_to?(:lstrip!) or flag_set.empty?
```
Comment (sanzmauro) [inline lib/splitclient-rb/cache/filter/flag_set_filter.rb:18]: why you are using `respond_to?` ??

File: lib/splitclient-rb/cache/filter/flag_set_filter.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+            return false
+          end
+
+          @flag_sets.intersection([flag_set]).any?
+        end
+
+        def intersect?(flag_sets)
+          if not @should_filter
+            return true
+          end
```
Comment (sanzmauro) [inline lib/splitclient-rb/cache/filter/flag_set_filter.rb:28]: same here

File: lib/splitclient-rb/cache/filter/flag_set_filter.rb (hunk 1 above)
Comment (chillaq) [reply lib/splitclient-rb/cache/filter/flag_set_filter.rb:17]: will do

File: lib/splitclient-rb/cache/filter/flag_set_filter.rb (hunk 3 above)
Comment (chillaq) [reply lib/splitclient-rb/cache/filter/flag_set_filter.rb:28]: will do

**Pull request closed without merging**